silhouette/src/main/scala/silhouette/http/client/RequestBuilder.scala

```diff
@@ -0,0 +1,81 @@
+/**
+ * Licensed to the Minutemen Group under one or more contributor license
+ * agreements. See the COPYRIGHT file distributed with this work for
+ * additional information regarding copyright ownership.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you
+ * may not use this file except in compliance with the License. You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package silhouette.http.client
+
+import scala.concurrent.Future
+
+/**
+ * Builds a request by providing a fluent interface.
+ *
+ * An instance of this class should not be instantiated directly. Instead the [[HttpClient]] should be
+ * used to instantiate the [[RequestBuilder]].
+ */
+private[silhouette] trait RequestBuilder {
+
+  /**
+   * The concrete implementation of this trait.
+   */
+  type Self <: RequestBuilder
+
+  /**
+   * Returns a copy of this instance with a new URL.
+   *
+   * @param url The URL to set.
+   * @return A request builder to provide a fluent interface.
+   */
+  def withUrl(url: String): Self
+
+  /**
+   * Returns a copy of this instance with a new HTTP method.
+   *
+   * @param method The HTTP method to set.
+   * @return A request builder to provide a fluent interface.
+   */
+  def withMethod(method: String): Self
+
+  /**
+   * Returns a copy of this instance with the list of headers set.
+   *
+   * @param params The headers to set.
+   * @return A request builder to provide a fluent interface.
+   */
+  def withHeader(params: Seq[(String, String)]): Self
```

Comment:
I think we should use the same signature as in https://github.com/minutemen/silhouette/blob/master/silhouette/src/main/scala/silhouette/http/RequestPipeline.scala#L108